Framework/Core/src/LifetimeHelpers.cxx

```diff
@@ -98,7 +99,11 @@ ExpirationHandler::Creator LifetimeHelpers::enumDrivenCreation(size_t start, siz
 
 ExpirationHandler::Creator LifetimeHelpers::timeDrivenCreation(std::chrono::microseconds period)
 {
-  auto start = getCurrentTime();
+  std::random_device r;
+  std::default_random_engine e1(r());
+  std::uniform_int_distribution<uint64_t> dist(0, period.count());
```

Comment:
I thought about it again, and perhaps we could use `(-period.count() / 2, period.count() / 2)` as random range?

Reply:
My  worry is that then `delta` below overflows and we end up with a timer in the future. Maybe the correct thing to do is: `-rand(0, period.count() / 2) - period.count() / 2` so that we still guarantee at least half of the period for the first timer duration, but we are still random enough to make it sustainable on the receiving side...

Reply:
but then you do not have proper randomization, you randomize only over half the period. I would not do that.

Reply:
Well, we can tweak that to include 3/4 of the period of something like that. I think triggering a QC task with a time in the future is more dangerous if the value of the timestamp is actually used e.g. for a running average or things like that.